src/pool/srv_pool.c

```diff
@@ -1810,6 +1810,11 @@ pool_svc_step_up_cb(struct ds_rsvc *rsvc)
 	} else {
 		uuid_generate(pool_hdl_uuid);
 		uuid_generate(cont_hdl_uuid);
+		/* Only copy server handle to make is_from_srv() check correctly, and
+		 * container server handle will not be copied here, otherwise
+		 * ds_pool_iv_refresh_hdl will not open the server handle container.
+		 */
+		uuid_copy(svc->ps_pool->sp_srv_pool_hdl, pool_hdl_uuid);
```

Comment:
By the way, there seems to be a data race on `sp_srv_pool_hdl` if a non-system xstream calls `ds_pool_iv_srv_hdl_fetch_non_sys` to read `sp_srv_pool_hdl` when the `pool_iv` code or this new `uuid_copy` code updates `sp_srv_pool_hdl`.

Reply:
sp_srv_pool_hdl and sp_srv_cont_hdl will only be accessed by system xstream. Hmm, ds_pool_iv_srv_hdl_fetch_non_sys is obsolete, should be removed.  let me remove it if the PR needs to refresh.

Reply:
OK, that's good.